pmtiles/server.go

```diff
@@ -284,83 +312,102 @@ func (server *Server) getMetadata(ctx context.Context, httpHeaders map[string]st
 	httpHeaders["Content-Type"] = "application/json"
 	return 200, httpHeaders, metadataBytes
 }
-
 func (server *Server) getTile(ctx context.Context, httpHeaders map[string]string, name string, z uint8, x uint32, y uint32, ext string) (int, map[string]string, []byte) {
-	rootReq := request{key: cacheKey{name: name, offset: 0, length: 0}, value: make(chan cachedValue, 1)}
+	status, headers, data, purgeEtag := server.getTileAttempt(ctx, httpHeaders, name, z, x, y, ext, "")
+	if len(purgeEtag) > 0 {
+		// file has new etag, retry once force-purging the etag that is no longer value
+		status, headers, data, _ = server.getTileAttempt(ctx, httpHeaders, name, z, x, y, ext, purgeEtag)
+	}
+	return status, headers, data
+}
+func (server *Server) getTileAttempt(ctx context.Context, httpHeaders map[string]string, name string, z uint8, x uint32, y uint32, ext string, purgeEtag string) (int, map[string]string, []byte, string) {
+	rootReq := request{key: cacheKey{name: name, offset: 0, length: 0}, value: make(chan cachedValue, 1), purgeEtag: purgeEtag}
 	server.reqs <- rootReq
 
 	// https://golang.org/doc/faq#atomic_maps
 	rootValue := <-rootReq.value
 	header := rootValue.header
 
 	if !rootValue.ok {
-		return 404, httpHeaders, []byte("Archive not found")
+		return 404, httpHeaders, []byte("Archive not found"), ""
 	}
 
 	if z < header.MinZoom || z > header.MaxZoom {
-		return 404, httpHeaders, []byte("Tile not found")
+		return 404, httpHeaders, []byte("Tile not found"), ""
```

Comment:
Should this be a 204?

Reply:
I chose 404 because if you are asking for a tile outside of the defined zoom range that feels more like "undefined" than "known 0 length tile" that a 204 would be, for example an empty tile within the bounds and zoom range.